src/systems/movement.js

```diff
@@ -51,6 +52,27 @@ const splatterBlood = (entity) => {
   });
 };
 
+const bumpAttack = (targetEntity) => {
+    if (targetEntity.health) {
+        hit(targetEntity);
+
+        if (targetEntity.health.current <= 0) {
+          kill(targetEntity);
+        }
+      }
+}
+
+const washInFountain = (targetEntity, fountain) => {
+    if (targetEntity.has('Soilage')) {
+        if (fountain.has('Soilage')) {
+            console.log("You can't clean yourself in this foul fountain")
+        } else {
+            targetEntity.get('Soilage').forEach(x => fountain.add('Soilage', { ...x.serialize() }))
```

Comment:
nice use of serialize here! any thoughts on a better interface experience for this use case? maybe a `clone()` function to components?

Reply:
Clone would be a nice addition yeah, a bit more obvious for the usecase. But I gotta say - geotic has been a joy to use - a lot of things clicked for me tonight working with events, so nice to be able to focus on the game and not the engine 🙌